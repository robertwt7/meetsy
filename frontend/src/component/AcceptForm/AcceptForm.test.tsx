import { screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { LocalizationProvider } from "@mui/lab";
import DateAdapter from "@mui/lab/AdapterDayjs";
import { FunctionComponent } from "react";
import MockDate from "mockdate";
import { AcceptForm } from "./AcceptForm";
import { render } from "../../testutils/test-utils";

MockDate.set("2020-01-01");

const mocks = {
  useSession: jest.fn().mockReturnValue({
    sessionData: {
      user: {
        name: "Test user",
        email: "test@gmail.com",
      },
    },
    status: "authenticated",
  }),
};

jest.mock("../SnackBar", () => ({
  useSnackBar: jest.fn().mockReturnValue(jest.fn()),
}));

jest.mock("../UserCalendar");

jest.mock("next/router", () => ({
  useRouter: jest.fn().mockReturnValue({ push: jest.fn() }),
}));

jest.mock("next-auth/react", () => ({
  useSession: () => mocks.useSession(),
}));

jest.mock("src", () => ({
  useGetInvitedEventQuery: jest.fn().mockReturnValue({
    data: {
      user: { userId: 1 },
      duration: 30,
      name: "Event",
      location: "Melbourne",
      notes: "Note",
      expiry: "2099-01-01T00:00:00.000Z",
      available_dates: [],
    },
    isFetching: false,
    isError: false,
    error: {},
  }),
}));

const Component: FunctionComponent = () => (
  <LocalizationProvider dateAdapter={DateAdapter}>
    <AcceptForm url="testing-my-url" />
  </LocalizationProvider>
);

describe("AcceptForm Component", () => {
  afterAll(() => {
    MockDate.reset();
  });

  test("should match snapshot", () => {
    const { baseElement } = render(<Component />);
    expect(baseElement).toMatchSnapshot();
  });
  test("should render url", () => {
    render(<Component />);
    expect(screen.getByText("Meeting Details")).toBeInTheDocument();
  });

  test("should render sign in button when unauthenticated", () => {
    mocks.useSession.mockReturnValue({
      sessionData: {
        user: {},
      },
      status: "unauthenticated",
    });
    render(<Component />);

    expect(screen.getByTestId("sign-in-button")).toBeInTheDocument();
  });
});
