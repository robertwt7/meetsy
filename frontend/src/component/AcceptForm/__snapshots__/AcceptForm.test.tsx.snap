// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcceptForm Component should match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex w-full flex-col py-4 md:py-0"
    >
      <form
        action="#"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 w-full css-1ps6pg7-MuiPaper-root"
        >
          <div
            class="flex flex-col md:flex-row md:items-stretch"
          >
            <div
              class="w-full bg-gray-50 md:w-1/3"
            >
              <div
                class="m-4 space-y-2"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 text-gray-800 css-ahj2mt-MuiTypography-root"
                >
                  Meeting Details
                </p>
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                >
                  Event
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  üïñ Duration: 
                  30
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  üìç Location: 
                  Melbourne
                </p>
                <div
                  class="flex flex-row items-center space-x-2"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    üëã From: 
                     
                     (
                    )
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  üìñ Notes: 
                  Note
                </p>
              </div>
              <div
                class="m-4 space-y-2"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 text-gray-800 css-ahj2mt-MuiTypography-root"
                >
                  Your Details
                </p>
                <div
                  class="flex flex-row items-center space-x-2"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    üëã Me: 
                     (
                    )
                  </p>
                </div>
              </div>
              <div
                class="m-4 mt-6 "
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 text-gray-800 css-ahj2mt-MuiTypography-root"
                >
                  Available Dates
                </p>
                <div
                  class="MuiCalendarPicker-root ml-0 mr-auto css-169iwlq-MuiCalendarPicker-root"
                >
                  <div
                    class="css-1dozdou"
                  >
                    <div
                      class="css-l0iinn"
                      role="presentation"
                    >
                      <div
                        class="PrivatePickersFadeTransitionGroup-root css-1bx5ylf"
                      >
                        <div
                          aria-live="polite"
                          class="css-1v994a0"
                          style="opacity: 1;"
                        >
                          January
                        </div>
                      </div>
                      <div
                        class="PrivatePickersFadeTransitionGroup-root css-1bx5ylf"
                      >
                        <div
                          aria-live="polite"
                          class="css-1v994a0"
                          style="opacity: 1;"
                        >
                          2020
                        </div>
                      </div>
                      <button
                        aria-label="calendar view is open, switch to year view"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-7kykdr-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fd2y78-MuiSvgIcon-root"
                          data-testid="ArrowDropDownIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      class="css-k008qs"
                      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                    >
                      <button
                        aria-label="Previous month"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1wgcwo1-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        title="Previous month"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="ArrowLeftIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                      <div
                        class="css-18puooo"
                      />
                      <button
                        aria-label="Next month"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeSmall css-nfrs8p-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        title="Next month"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="ArrowRightIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="PrivatePickersFadeTransitionGroup-root MuiCalendarPicker-viewTransitionContainer css-1wvgxus-MuiCalendarPicker-viewTransitionContainer"
                  >
                    <div
                      style="opacity: 1;"
                    >
                      <div
                        class="css-1n2mv2k"
                      >
                        <span
                          aria-hidden="true"
                          class="MuiTypography-root MuiTypography-caption css-1w13o7u-MuiTypography-root"
                        >
                          S
                        </span>
                        <span
                          aria-hidden="true"
                          class="MuiTypography-root MuiTypography-caption css-1w13o7u-MuiTypography-root"
                        >
                          M
                        </span>
                        <span
                          aria-hidden="true"
                          class="MuiTypography-root MuiTypography-caption css-1w13o7u-MuiTypography-root"
                        >
                          T
                        </span>
                        <span
                          aria-hidden="true"
                          class="MuiTypography-root MuiTypography-caption css-1w13o7u-MuiTypography-root"
                        >
                          W
                        </span>
                        <span
                          aria-hidden="true"
                          class="MuiTypography-root MuiTypography-caption css-1w13o7u-MuiTypography-root"
                        >
                          T
                        </span>
                        <span
                          aria-hidden="true"
                          class="MuiTypography-root MuiTypography-caption css-1w13o7u-MuiTypography-root"
                        >
                          F
                        </span>
                        <span
                          aria-hidden="true"
                          class="MuiTypography-root MuiTypography-caption css-1w13o7u-MuiTypography-root"
                        >
                          S
                        </span>
                      </div>
                      <div
                        class="PrivatePickersSlideTransition-root css-dhopo2"
                      >
                        <div
                          class="css-i6bazn"
                          role="grid"
                        >
                          <div
                            class="css-mvmu1r"
                            role="row"
                          >
                            <div
                              role="cell"
                            >
                              <div
                                class="MuiPickersDay-root MuiPickersDay-dayWithMargin MuiPickersDay-hiddenDaySpacingFiller css-188yni7-MuiPickersDay-root"
                              />
                            </div>
                            <div
                              role="cell"
                            >
                              <div
                                class="MuiPickersDay-root MuiPickersDay-dayWithMargin MuiPickersDay-hiddenDaySpacingFiller css-188yni7-MuiPickersDay-root"
                              />
                            </div>
                            <div
                              role="cell"
                            >
                              <div
                                class="MuiPickersDay-root MuiPickersDay-dayWithMargin MuiPickersDay-hiddenDaySpacingFiller css-188yni7-MuiPickersDay-root"
                              />
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 1, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root Mui-selected MuiPickersDay-dayWithMargin MuiPickersDay-today css-195y93z-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                1
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 2, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                2
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 3, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                3
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 4, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                4
                              </button>
                            </div>
                          </div>
                          <div
                            class="css-mvmu1r"
                            role="row"
                          >
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 5, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                5
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 6, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                6
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 7, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                7
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 8, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                8
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 9, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                9
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 10, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                10
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 11, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                11
                              </button>
                            </div>
                          </div>
                          <div
                            class="css-mvmu1r"
                            role="row"
                          >
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 12, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                12
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 13, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                13
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 14, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                14
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 15, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                15
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 16, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                16
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 17, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                17
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 18, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                18
                              </button>
                            </div>
                          </div>
                          <div
                            class="css-mvmu1r"
                            role="row"
                          >
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 19, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                19
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 20, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                20
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 21, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                21
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 22, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                22
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 23, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                23
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 24, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                24
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 25, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                25
                              </button>
                            </div>
                          </div>
                          <div
                            class="css-mvmu1r"
                            role="row"
                          >
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 26, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                26
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 27, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                27
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 28, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                28
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 29, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                29
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 30, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                30
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <button
                                aria-label="Jan 31, 2020"
                                class="MuiButtonBase-root Mui-disabled MuiPickersDay-root MuiPickersDay-dayWithMargin css-bkrceb-MuiButtonBase-root-MuiPickersDay-root"
                                disabled=""
                                tabindex="-1"
                                type="button"
                              >
                                31
                              </button>
                            </div>
                            <div
                              role="cell"
                            >
                              <div
                                class="MuiPickersDay-root MuiPickersDay-dayWithMargin MuiPickersDay-hiddenDaySpacingFiller css-188yni7-MuiPickersDay-root"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="m-4 mt-0"
                >
                  <div
                    class="flex flex-col"
                  >
                    <div
                      class="my-2"
                    >
                      <div
                        aria-invalid="false"
                        class="MuiFormControl-root MuiTextField-root w-full css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                      >
                        <label
                          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                          data-shrink="false"
                          for="mui-1"
                          id="mui-1-label"
                        >
                          Add your notes
                        </label>
                        <div
                          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <textarea
                            aria-invalid="false"
                            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                            id="mui-1"
                            name="notes"
                            rows="4"
                            style="height: 0px; overflow: hidden;"
                          />
                          <textarea
                            aria-hidden="true"
                            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                            readonly=""
                            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                            tabindex="-1"
                          />
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-173wfxe"
                            >
                              <span>
                                Add your notes
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="mt-8 w-full"
                    >
                      <button
                        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root w-full css-sghohy-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="submit"
                      >
                        Confirm
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="relative flex w-full flex-col overflow-auto border-l border-gray-300 p-4 md:w-2/3"
            >
              <div
                data-testid="user-calendar"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</body>
`;
